File: LinkedList/DeleteNodeWithVal/deleteVal.cpp
#include <iostream>
#include <climits>

struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode* farzi = new ListNode(INT_MAX);
        ListNode* dummy = farzi;
        ListNode* temp = head;
        while(temp!=NULL){
            int value = temp->val;
            if(value!=val){
                ListNode* help = new ListNode(value);
                farzi->next = help;
                farzi = farzi->next;
            }
            temp = temp->next;
        }
        return dummy->next;
    }
};

File: LinkedList/DeleteNodeWithVal/deleteVal.py
from typing import Optional
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        if not head: return None
        ans = None
        temp = None
        iterator = head
        while iterator:
            if iterator.val != val:
                if not ans:
                    if not temp:
                        temp = iterator
                    ans = temp
                else:
                    temp.next = iterator
                    temp = temp.next
            iterator = iterator.next
        if temp:
            temp.next = None
        return ans